src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/RazorFormattingPass.cs

```diff
@@ -86,11 +86,172 @@ private static IEnumerable<TextEdit> FormatRazor(FormattingContext context, Razo
                 // being overly careful to only try to format syntax forms they care about.
                 TryFormatCSharpCodeBlock(context, edits, source, node);
                 TryFormatSingleLineDirective(context, edits, source, node);
+                TryFormatTransitionCodeBlock(context, edits, source, node);
             }
 
             return edits;
         }
 
+        private static void TryFormatTransitionCodeBlock(FormattingContext context, IList<TextEdit> edits, RazorSourceDocument source, SyntaxNode node)
+        {
+            Range? openBraceRange = null;
+            Range? codeRange = null;
+            Range? closeBraceRange = null;
+            SyntaxNode? openBraceNode = null;
+            SyntaxNode? codeNode = null;
+            SyntaxNode? closeBraceNode = null;
+
+            // TODO: Split the if's into their own methods
+
+            // complex situations like
+            // @{
+            //  public void Method(){
+            //      @(DateTime.Now)
+            //  }
+            // }
+            if (node is CSharpRazorBlockSyntax csharpRazorBlock &&
+                csharpRazorBlock.Parent is CSharpCodeBlockSyntax firstCodeBlock &&
+                csharpRazorBlock.Parent.Parent is CSharpCodeBlockSyntax secondCodeBlock)
```

Comment:
`secondCodeBlock` in this case contains the `{` and `}` whereas `firstCodeBlock` is a thin wrapper around `csharpRazorBlock`. Those are obviously terrible names, but I'm not sure what the language name for `@ { ... }` is, since that would be my preferred name (@davidwengier's suggestion will certainly be an improvement anyway).

Basically since `CSharpCodeBlockSyntax` works overtime being applied to so many different contexts I'm trying to reconstitute the language concept we're dealing with. If the compiler defined us a `RazorCSharpThisThingySyntax` I could trim it down to just that.

Reply:
I see, ya ultimately a more well-formed syntax tree would save us. Love that this PR is really raising how we have to inspect the SyntaxTree